modules/unsupported/jdbc-singlestore/src/main/java/org/geotools/data/singlestore/SingleStoreDialect.java

```diff
@@ -0,0 +1,398 @@
+
+    public SingleStoreDialect(JDBCDataStore dataStore) {
+        super(dataStore);
+    }
+
+    @Override
+    public boolean includeTable(String schemaName, String tableName, Connection cx) throws SQLException {
+        if ("geometry_columns".equalsIgnoreCase(tableName)) {
+            return false;
+        }
+        return super.includeTable(schemaName, tableName, cx);
+    }
+
+    @Override
+    public String getNameEscape() {
+        return "";
+    }
+
+    @Override
+    public String getGeometryTypeName(Integer type) {
+        return super.getGeometryTypeName(type);
+    }
+
+    @Override
+    public Integer getGeometrySRID(String schemaName, String tableName, String columnName, Connection cx)
+            throws SQLException {
+        String sql = String.format(
+                "SELECT COLUMN_NAME, DATA_TYPE " + "FROM INFORMATION_SCHEMA.COLUMNS "
+                        + "WHERE TABLE_NAME = '%s' AND COLUMN_NAME = '%s'",
+                tableName, columnName);
+
+        dataStore.getLogger().fine(sql);
+
+        Statement st = cx.createStatement();
+        try {
+            ResultSet rs = st.executeQuery(sql);
+            try {
+                if (rs.next()) {
+                    return 4326;
+                }
+            } finally {
+                dataStore.closeSafe(rs);
+            }
+        } catch (SQLException e) {
+            // geometry_columns does not exist
+        } finally {
+            dataStore.closeSafe(st);
+        }
+        return null;
+    }
+
+    @Override
+    public void encodeGeometryEnvelope(String tableName, String geometryColumn, StringBuffer sql) {
+        encodeColumnName(null, geometryColumn, sql);
+    }
+
+    @Override
+    public Envelope decodeGeometryEnvelope(ResultSet rs, int column, Connection cx) throws SQLException, IOException {
+        try {
+            Geometry geom = new WKTReader().read(rs.getString(column));
+            return geom.getEnvelopeInternal();
+        } catch (ParseException e) {
+            String msg = "Error decoding wkb for envelope";
+            throw new IOException(msg, e);
+        }
+    }
+
+    @Override
+    public Geometry decodeGeometryValue(
+            GeometryDescriptor descriptor,
+            ResultSet rs,
+            String name,
+            GeometryFactory factory,
+            Connection cx,
+            Hints hints)
+            throws IOException, SQLException {
+        byte[] bytes = rs.getBytes(name);
+        if (bytes == null) {
+            return null;
+        }
+        try {
+            return new WKBReader(factory).read(bytes);
+        } catch (ParseException e) {
+            String msg = "Error decoding wkb";
+            throw (IOException) new IOException(msg).initCause(e);
+        }
+    }
+
+    @Override
+    public void registerClassToSqlMappings(Map<Class<?>, Integer> mappings) {
+        super.registerClassToSqlMappings(mappings);
+
+        //        mappings.put(Point.class, POINT);
```

Comment:
same as above. I would remove commented out pieces about types that are not used. Please also check other methods too.